### PR TITLE
Proper upgarde to Humble with fixed typos & dependencies

### PR DESCRIPTION
**Pull request by macmacal**
## Description
**Fixes:**
- missing dependencies (`xarco`, `ros2_control`)
- corrected all typos from `ati_axia` to `ati_axia80` (probably this should be revisited...),
- corrected arguments to launch node `controller_manager` in the launch file (from now the `robot_description` must be given directly from the topic),
- introduced a simple assert in `net_ft_interface.hpp` to explain the _Segmentation Fault_ from attempting to retrieve a non-existent factory from the `std::map<std::string, NetFTFactory*>`,

**Changes:**
- introduced pre-commit hook to autoformat the XML (both `.xarco` and `package.xml`) files,
- derived the launch flag `use_physical_hardware` to the default launch file for testing convince,
- defined missing C++ standard as version 17 (in the `CMakeLists.txt`),

## Motivation and Context

- This PR was created to fix all bugs regarding launching the package in the Humble distribution. 
- Resolves issue #2

## How Has This Been Tested?

Manually, launching the driver with the physical ATI AXIA80 ft sensor.

Used a docker container with RT capabilities:
```yaml
cap_add:
     - SYS_NICE
    ulimits:
      rtprio: 99
      rttime: -1
      memlock: 8428281856
    network_mode: host
```

## Checklist:

- [x] All TODOs in the code has been resolved or linked with a proper issue.
- [x] Code is (auto)formatted.
- [x] Documentation (README/CHANGELOG/Wiki/etc) has been updated.
- [ :cactus: ] All automatic checks passed.
    - The building stage is also broken in the main repository - it should be fixed in other PR. 

---
RoboLab ClickUp related task: [869619ebg](https://app.clickup.com/t/869619ebg)

### CODE REVIEW

File: net_ft_driver/include/net_ft_driver/interfaces/net_ft_interface.hpp [hunk 1]
```diff
@@ -72,9 +73,16 @@ class NetFTInterface
...
   static std::unique_ptr<NetFTInterface> create(const std::string& sensor_type, const std::string& ip_address)
   {
+    assert_sensor_type(sensor_type);
     return get_factory_instance()[sensor_type]->create(ip_address);
   }
 
+  static void assert_sensor_type(const std::string& sensor_type)
+  {
+    if (get_factory_instance().find(sensor_type) == get_factory_instance().end())
+      throw std::runtime_error("Sensor type not supported: " + sensor_type);
```
Comment (macmacal) [inline net_ft_driver/include/net_ft_driver/interfaces/net_ft_interface.hpp:83]: Consider to improve the verbosity of the output for the end user. Currently it looks like this:

image

File: README.md [hunk 2]
```diff
@@ -41,13 +45,14 @@ source install/local_setup.sh
 Launch the controller:
 
 ```Bash
```
Comment (sivral) [inline README.md]: Please be consistent with the markdown.

**Review by sivral — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (macmacal) [reply README.md]: done

File: net_ft_driver/include/net_ft_driver/interfaces/net_ft_interface.hpp (hunk 1 above)
Comment (macmacal) [reply net_ft_driver/include/net_ft_driver/interfaces/net_ft_interface.hpp:83]: For now its just ~one liner, good enough to be in the `.hpp` file. Also, there is no obvious way to get the logger.

**Review by sivral — APPROVED** [review]

**Review by macmacal — COMMENTED** [review]
LGTM

**Pull request merged**